### PR TITLE
Log validation score

### PR DESCRIPTION
**Pull request by turian**

### CODE REVIEW

File: heareval/predictions/task_predictions.py [hunk 1]
```diff
@@ -782,14 +783,15 @@ def task_predictions_train(
...
-            "time_in_min %.2f" % ((end - start) / 60),
+            "time_in_min %.2f" % time_in_min,
             json.dumps(hparams_to_json(predictor.hparams)),
             "\n\n\n",
         )
         sys.stdout.flush()
         return (
             checkpoint_callback.best_model_path,
             epoch,
+            time_in_min,
```
Comment (jorshi) [inline heareval/predictions/task_predictions.py:794]: This return is starting to get a bit gnarly. Does it have to be a list? getting to the point where returning a custom object or a dictionary would be nice

File: heareval/predictions/task_predictions.py [hunk 2]
```diff
@@ -897,22 +900,31 @@ def print_scores(mode, scores_and_trainers):
...
+        for score, _, _, _, _, _, postprocessing in scores_and_trainers:
             w.write(json.dumps([score, postprocessing]) + "\n")
 
     # Use that model to compute test scores
     (
         best_score,
         best_model_path,
-        best_model_epoch,
+        best_epoch,
+        best_time_in_min,
```
Comment (jorshi) [inline heareval/predictions/task_predictions.py:927]: I might be reading this wrong, but why it scores_and_trainers[0] the best? Isn't it just the first grid point? I'm not seeing where scores_and_trainers is sorted based on scores before this selection

File: heareval/predictions/task_predictions.py (hunk 1 above)
Comment (turian) [reply heareval/predictions/task_predictions.py]: @jorshi check latest push

File: heareval/predictions/task_predictions.py [hunk 3]
```diff
@@ -954,19 +948,32 @@ def print_scores(mode, scores_and_trainers):
...
+            "score_mode": best_grid_point.score_mode,
+            "embedding_path": embedding_path,
+        }
     )
-    assert len(test_scores) == 1, "Should have only one test dataloader"
-    test_scores = test_scores[0]
-
     open(embedding_path.joinpath("test.predicted-scores.json"), "wt").write(
-        json.dumps(test_scores, indent=4)
+        json.dumps(test_results, indent=4)
```
Comment (jorshi) [inline heareval/predictions/task_predictions.py:974]: Something in here isn't json serializable

File: heareval/predictions/task_predictions.py [hunk 4]
```diff
@@ -789,30 +809,21 @@ def task_predictions_train(
...
-        sys.stdout.flush()
-        return (
-            checkpoint_callback.best_model_path,
-            epoch,
-            dict(predictor.hparams),
-            best_postprocessing,
-            trainer,
-            checkpoint_callback.best_model_score.detach().cpu().item(),
-            mode,
+        return GridPointResult(
```
Comment (jorshi) [inline heareval/predictions/task_predictions.py:818]: Beautiful

File: heareval/predictions/task_predictions.py (hunk 3 above)
Comment (turian) [reply heareval/predictions/task_predictions.py:975]: looking now

**Review by jorshi — APPROVED** [review]

**Pull request merged**